### ISSUE TITLE
Script with prebid Server (S2S) and conditional ad units doesn't works

### ISSUE THREAD

**Issue by javipm** [prebid/Prebid.js#2222]
## Type of issue
bug

## Description
I've created a script with Prebid 1.0 and conditional ad units (http://prebid.org/dev-docs/conditional-ad-units.html) from the browser side. The script works but when the S2S is activated, the process doesn't work and the debug says:

`callBids executed with no bidRequests. Were they filtered by labels or sizing?
`
## Steps to reproduce

I've created a script with Prebid 1.0 using the conditional ad units (http://prebid.org/dev-docs/conditional-ad-units.html).

I've labels for the screen size:

```
                 const customSizeConfig = [{
			'mediaQuery': '(min-width: 1024px)',
			'sizesSupported': [
				[980, 90],
				[728, 90],
				[980, 250],
				[970, 90],
				[970, 250],
				[990, 90],
				[160, 600],
				[120, 600],
				[120, 1000],
				[120, 800],
				[300, 250],
				[300, 300],
				[300, 600],
				[400, 300],
				[640, 360],
				[640, 480]
			],
			'labels': ['desktop']
		}, {
			'mediaQuery': '(min-width: 768px) and (max-width: 1023px)',
			'sizesSupported': [
				[728, 90],
				[300, 250],
				[300, 600],
				[160, 600],
				[400, 300],
				[640, 360],
				[640, 480]
			],
			'labels': ['tablet']
		}, {
			'mediaQuery': '(max-width: 767px)',
			'sizesSupported': [
				[320, 50],
				[320, 100],
				[300, 100],
				[300, 250],
				[300, 600],
				[300, 300],
				[400, 300],
				[640, 360],
				[640, 480]
			],
			'labels': ['mobile']
		}];
```

Also a personalized label to differentiate in which part of the page we are:

```
                 if (document.body.classList.contains('home')) {
 			customLabel = ['portada'];
 		}
```
 

```
	                 pbjs.setConfig({
				priceGranularity: customGranularity,
				setBidderSequence: 'random',
				sizeConfig: customSizeConfig,
			});

			pbjs.addAdUnits(adUnits);

			pbjs.requestBids({
				labels: customLabel,
				bidsBackHandler: sendAdserverRequest,
			});
```

It is necessary that both tags are active to activate the adunits.

The script works well from the browser side, but by activating the S2S with this code:

```
			pbjs.setConfig({
				priceGranularity: customGranularity,
				setBidderSequence: 'random',
				sizeConfig: customSizeConfig,
				s2sConfig: {
			            accountId: 'xxx',
			            enabled: true,
			            bidders: ['appnexus', 'rubicon'],
			            defaultVendor: 'appnexus'
			        },
			});

```
does not make any request, throws the following warning:

`callBids executed with no bidRequests. Were they filtered by labels or sizing?
`

Is the S2S compatible with conditional ad units? Thanks

## Test page

- Browser side with conditionals ad units (works): http://cdn.sibboventures.net/github/browser.html?pbjs_debug=true

- S2S with conditional ad units (doesn't work):  http://cdn.sibboventures.net/github/s2s.html?pbjs_debug=true

- S2S without conditional ad units (works): http://cdn.sibboventures.net/github/s2s_noconditional.html?pbjs_debug=true

### Expected results

Script with S2S enabled and conditional ad units make requests

### Actual results

Script with S2S enabled and conditional ad units doesn't make requests

## Platform details
Prebid 1.5.0-pre
I've also tested with 1.4.0 with the same result

**Comment by javipm** [comment]
Hi

I've found some possible causes:

- In adaptermanager.js, in the getAdUnitCopyForPrebidServer function, the sizes are converted to the prebidServer format:

    `adUnit.sizes = transformHeightWidth (adUnit);`

This format is not compatible with the size array needed by resolveStatus of sizeMapping.js

- In prebidServerBidAdapter.js, the adUnits aren't being used with the sizes filtered by sizeMapping. The adUnits of the s2sBidRequest variable are used instead of the bidRequests variable, which is where they are filtered.

Thanks

**Comment by mercuryyy** [comment]
Had the same issue today, with some research i found your post after i posted one of my own - https://github.com/prebid/Prebid.js/issues/2315

Looks like this issue was first reported 18 days ago and now stale.

@mkendall07 @prebid @BartVB  

Can you kindly take a look ?

**Comment by jsnellbaker** [comment]
@javipm  This should be fixed now in 1.7 release.  If you continue to see any issues, please let us know.
